src/src/dynpower.cpp

```diff
@@ -195,4 +196,33 @@ void DynamicPower_Update(uint32_t now)
   }
 }
 
-#endif // TARGET_TX
\ No newline at end of file
+#endif // TARGET_TX
+
+#if defined(TARGET_RX)
+
+/***
+ * @brief: Set power to configured power or update power to match the current TX power
+ * @param initialize: Set to true if calling from setup() to force even PWR_MATCH_TX to default power
+*/
+void DynamicPower_UpdateRx(bool initialize)
```

Comment:
this boolean argument can be removed if the method is split into two.

```
void DynamicPower_InitializeRx()
```

and

```
void DynamicPower_UpdateRx(void)
```

Boolean arguments are generally a bad idea and often indicate there should be two separate methods.

Random article on the subject:

https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3

Reply:
I would agree but I'm more concerned with the duplication of code. The code of `DynamicPower_InitializeRx()` would duplicate the PWR_MATCH_TX check and set power to config power code and that's more important to me: not duplicating the same logic in two places. Someone might change the logic in one and forget to change the logic in the other and later lead to a bug. Having it all in one function with no duplicated logic makes it easier to understand from that perspective.

You could then argue that that code should be factored out into another function that returns true if it set the power and call that from both IntializaRx and UpdateRx, but then you'd need to see the implementation to know in what instance it returns true, which was your complaint about the bool parameter except in result form. That also generally would create more binary code adding an additional call to the new function and compare against the result in two places.

Having 2 functions and having to always drill down to the third function to see how to expect them to behave certainly makes each more digestible but I find drilling down layers deep to be harder to follow, even if each layer is 1 line of code. It could be a good pattern if there were a substantial amount of distinct code in every function, but this is 4 lines of code with 3 conditions but that does not meet my threshold for turning 1 function into 3.

Reply:
yeah, everything's always a compromise, I'm happy you took the time to review it and for your detailed reply and consideration of my observation. 😍

Personally I try to never write them in the first place, then I don't even have to think about the odd instruction or binary size increase and hope the compiler and optimizer does a good job, which in 2023 they often do.

With regard to performance, you have the option of marking methods or blocks /hot/ or /cold/ so the optimizer can optimize for the /hot/ case which is where it's called with 'false' in the inner loop.

anyway, that's probably enough conversation on this!  was just a minor observation.